File: server/persistence/src/main/java/entity/Event.java
package entity;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
@ToString
@Builder
@Entity
public class Event implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer idEvent;

    @Column
    private String titleEvent;

    @Column
    private String overview;

    @ManyToMany(targetEntity = User.class)
    @JoinTable(name = "event_enrolledUsers", joinColumns = @JoinColumn(name = "idEvent"),
            inverseJoinColumns = @JoinColumn(name = "idUser"))
    private List<User> enrolledUsers;

    @ManyToOne
    @JoinColumn(name = "idUser")
    private User contactPerson;

    @Column
    private Integer maxEnrolledUsers;

    @ManyToOne
    @JoinColumn(name = "idLocation")
    private Location location;
    @Column
    private String otherLocation;

    @ManyToOne
    @JoinColumn(name = "idMeetingHall")
    private MeetingHall meetingHall;

    @Column
    private String keywords;

    @Column
    private Date eventDate;

    @Column
    private String eventTime;
}

File: client/angular-front/src/app/domain/event.ts
import {User} from './user';
import {Location} from './location';

export class Event {
  idEvent: number;
  titleEvent: string;
  overview: string;
  keywords: string;
  contactPerson: User;
  enrolledUsers: User[];
  maxEnrolledUsers: number;
  location: Location;
  stringDate: string;
  eventDate: Date;
  eventTime: string;
  placesLeft: number;
  meetingHall: MeetingHall;
  isUserEnrolled: Boolean;
  otherLocation: string;
}

export class MeetingHall {
  idMeetingHall: number;
  hallName: string;
  capacity: number;
  location: Location;
  floor: number;
}
